src/plugin.ts

```diff
@@ -98,23 +98,35 @@ function topicsToArray(input: any, base?: string): Topic[] {
   })
 }
 
-// eslint-disable-next-line valid-jsdoc
+// essentially just "cd .."
+function * up(from: string) {
+  while (path.dirname(from) !== from) {
+    yield from
+    from = path.dirname(from)
+  }
+  yield from
+}
+
+async function findSourcesRoot(root: string) {
+  for (const next of up(root)) {
+    const cur = path.join(next, 'package.json')
+    // eslint-disable-next-line no-await-in-loop
+    if (await exists(cur)) return path.dirname(cur)
+  }
+}
+
 /**
+ * @returns string
+ * @param name string
+ * @param root string
  * find package root
  * for packages installed into node_modules this will go up directories until
  * it finds a node_modules directory with the plugin installed into it
  *
- * This is needed because of the deduping npm does
+ * This is needed because of some oclif plugins do not declare the `main` field in their package.json
+ * https://github.com/oclif/config/pull/289#issuecomment-983904051
  */
-async function findRoot(name: string | undefined, root: string) {
-  // essentially just "cd .."
-  function * up(from: string) {
-    while (path.dirname(from) !== from) {
-      yield from
-      from = path.dirname(from)
-    }
-    yield from
-  }
+async function findRootLegacy(name: string | undefined, root: string): Promise<string | undefined> {
```

Comment:
> @benatshippabo Good find!
> 
> Unfortunately, if we require that plugins must specify the `main` field then this would be a breaking change, which we're not willing to do on this package since it will soon be in maintenance mode.

That's unfortunate, maybe we can see these changes once the package that will supersede this one is released?

> Is it possible to fallback to the old behavior whenever a module isn't found?

Done! @mdonnalley I re-added the legacy dependency resolution function as a fallback
https://github.com/benatshippabo/oclif-config/blob/12f8d90adb6c2d2ff632d37a0c505cf5583d1290/src/plugin.ts#L129-L147